pkg/ottl/expression.go

```diff
@@ -69,10 +71,58 @@ func (l literal[K]) Get(context.Context, K) (interface{}, error) {
 
 type exprGetter[K any] struct {
 	expr Expr[K]
+	keys []Key
 }
 
 func (g exprGetter[K]) Get(ctx context.Context, tCtx K) (interface{}, error) {
-	return g.expr.Eval(ctx, tCtx)
+	result, err := g.expr.Eval(ctx, tCtx)
```

Comment:
Testing the error conditions here as well would probably be good.

Reply:
Good idea